src/main/resources/templates/device_error.html

```diff
@@ -37,13 +37,5 @@ <h1 th:text="#{notAuthenticated}"></h1>
 
 </div>
 </body>
-<script>
-
-    function switchLang(lang) {
-        const searchParams = new URLSearchParams(window.location.search);
-        searchParams.set("lang", lang);
-        window.location.search = searchParams.toString();
-    }
-
-</script>
-</html>
\ No newline at end of file
+<script th:src="@{/lang.js}" type="text/javascript"></script>
```

Comment:
Nitpick, the type is to be left off if the content is javascript (see e.g. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script)